qt/python/mantidqt/widgets/saveprojectdialog/presenter.py

```diff
@@ -0,0 +1,47 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+#  This file is part of the mantid workbench
+
+from qtpy.QtWidgets import QDialogButtonBox
+
+from mantidqt.widgets.saveprojectdialog.view import ProjectSaveDialogView
+from workbench.config import CONF
+
+
+class ProjectSaveDialogPresenter:
+
+    def __init__(self, project, view=None):
+        self.view = view if view else ProjectSaveDialogView()
+
+        self.project = project
+
+        self.view.browse_push_button.clicked.connect(self.browse_button_clicked)
+        self.view.location_line_edit.textChanged.connect(self.location_selected)
+        self.view.accepted.connect(self.save_as)
+        self.view.rejected.connect(self.cancel)
+
+        self.view.set_save_altered_workspaces_only(project.save_altered_workspaces_only)
+        self.view.set_location(project.last_project_location)
+        self.view.buttonBox.button(QDialogButtonBox.Ok).setEnabled(self.view.get_location() != "")
+        self.view.exec()
+
+    def location_selected(self):
+        self.view.buttonBox.button(QDialogButtonBox.Ok).setEnabled(True)
```

Comment:
We are aiming to keep our presenters free of Qt so it's clear they are for presentation logic only. Could we add a `set_ok_enabled(bool)` method to the view and call that?